### PR TITLE
Speaking test module (add-countdown-timer)

### PR DESCRIPTION
**Pull request by dimahalaiskiy**
[](url)

### CODE REVIEW

**Comment by dimahalaiskiy** [comment]
https://user-images.githubusercontent.com/74025448/126782243-544c0367-cc9a-4aab-ac11-085d2258bd6d.mov

File: src/app/models/speaking-test/speaking-test.component.ts [hunk 1]
```diff
@@ -22,6 +22,15 @@ export class SpeakingTestComponent implements OnInit {
 
   counter = 0;
 
+  // for countdpown timer
```
Comment (RomPavel) [inline src/app/models/speaking-test/speaking-test.component.ts:25]: typo countdown

File: src/app/models/speaking-test/speaking-test.component.ts [hunk 2]
```diff
@@ -22,6 +22,15 @@ export class SpeakingTestComponent implements OnInit {
...
+  intervalId = 0;
+
+  message = '';
```
Comment (RomPavel) [inline src/app/models/speaking-test/speaking-test.component.ts:28]: do you use it?

File: src/app/models/speaking-test/speaking-test.component.ts [hunk 3]
```diff
@@ -22,6 +22,15 @@ export class SpeakingTestComponent implements OnInit {
...
+
+  minutes = 4;
```
Comment (RomPavel) [inline src/app/models/speaking-test/speaking-test.component.ts:30]: did you try to have 5 minutes 0 sec as defult?

File: src/app/models/speaking-test/speaking-test.component.ts [hunk 4]
```diff
@@ -55,6 +64,38 @@ export class SpeakingTestComponent implements OnInit {
...
+      if (this.minutes === 0 && this.seconds === 0) {
+        this.toggleRecording();
+      } else if (this.seconds === 0) {
+        this.minutes -= 1;
+        this.seconds = 59;
+      }
+    }, 1000);
+  }
+
+  // finish coundown timer
```
Comment (RomPavel) [inline src/app/models/speaking-test/speaking-test.component.ts:97]: remove this

**Review by Inunor — APPROVED** [review]

File: src/app/models/speaking-test/speaking-test.component.ts (hunk 2 above)
Comment (dimahalaiskiy) [reply src/app/models/speaking-test/speaking-test.component.ts:28]: Yes I do, Its already repair

**Pull request merged**